### PR TITLE
Keep state during hot reload, Webpack 2

### LINKED ISSUE #1: Loosing internal state when doing hot reload
```ts
// src/App.tsx
import * as React from "react";

export default class App extends React.Component<any, any> {
	private timer;
	private state = {
		counter: 0
	}
	constructor(prop) {
		super(prop)
		this.timer = setInterval(()=>{
			
		this.setState({counter: this.state.counter + 10})
		},1000)
		console.log("timer", this.timer )
		
	}
    render() {
        return (
            <div>
                <h1>Hello world!</h1>
                <div>Welcome to hot-reloading React written in TypeScript! {this.state.counter}</div>
            </div>
        );
    }
}
```

When I change the behavior of `this.state.counter + 10` to `this.state.counter - 10` for example, after module reload, the internal state get reset to `0`. Is there something we can further configure, or is it because of (already closed bug) https://github.com/gaearon/react-hot-loader/issues/276?

### PR DESCRIPTION
**Pull request by danzel**
Fixes #1 

I've basically redone #2, but with all the latest stuff.
There is now also a minimal example of state updates in app.tsx.

I've removed `server.js` and replaced it with running `webpack-dev-server` direct (More personal preference than anything)

### CODE REVIEW

**Comment by cr0cK** [comment]
Up!

This should be merged, I fixed the same things during 1 hour, I wanted to do a PR when I see this one. :neutral_face:

**Comment by Glavin001** [comment]
Hey! Thanks for commenting @cr0cK. I was going to merge #2 however I could not get it to work. I will try this new PR now and merge ASAP!
Thank you @danzel for contributing!

File: README.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 > Demo of React Hot Reloading + TypeScript
 
-```bash
-npm start
+## Install prerequisites
+```
+npm install -g webpack webpack-dev-server
```
Comment (Glavin001) [inline README.md:7]: Don't need this so long as:
1) they are in dependencies of package.json
2) we use the `scripts` from package.json, such as `npm start`

File: README.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
+npm install
+```
+
+## Usage
+```
+webpack-dev-server
```
Comment (Glavin001) [inline README.md:13]: Change to `npm start`

**Review by Glavin001 — APPROVED** [review]

**Pull request merged**